src/targets/stm32f1-generic/main.c

```diff
@@ -51,9 +56,33 @@ int main()
 
 	gpio_apply_config(gpio_config);
 
+	struct hid_hal_t hid_hal;
+	u8 info_functions[] = {
+		OI_FUNCTION_VERSION,
+		OI_FUNCTION_FW_INFO,
+		OI_FUNCTION_SUPPORTED_FUNCTION_PAGES,
+		OI_FUNCTION_SUPPORTED_FUNCTIONS,
+	};
+
+	/* create protocol config */
+	memset(&protocol_config, 0, sizeof(protocol_config));
+	protocol_config.device_name = "openinput stm32f1 Device";
+	protocol_config.hid_hal = hid_hal_init();
+	protocol_config.functions[INFO] = info_functions;
+	protocol_config.functions_size[INFO] = sizeof(info_functions);
+
 	usb_init();
 
 	for (;;) {
 		tud_task();
 	}
 }
+
+/* Invoked when received SET_REPORT control request */
```

Comment:
Currently, we are using the interrupt pipe to send and receive the data instead of the SET_REPORT and GET_REPORT requests. I have not documented this properly, sorry.

Reply:
Yes, i know, but this is how the tiny usb stack handles the incoming interrups transfers, if i ubderstood it correctly (this in particular is not well documented in the stack either)

Reply:
That comment is somewhat missplaced there, as it doesnt properly represent what is happening